### PR TITLE
feat: add weightPriority option

### PR DESCRIPTION
**Pull request by daybrush**
#2

### CODE REVIEW

File: src/grids/PackingGrid.ts [hunk 1]
```diff
@@ -199,6 +206,32 @@ export class PackingGrid extends Grid<PackingGridOptions> {
...
+      ratioWeight,
+    } = this.options;
+
+    if (weightPriority === "size") {
+      return 1;
+    } else if (weightPriority === "ratio") {
+      return 100;
+    }
+    return ratioWeight;
+  }
```
Comment (mixed) [inline src/grids/PackingGrid.ts]: It looks almost the same code, but is there any need to improve it?

**Review by mixed — COMMENTED** [review]
That's good code. _getSizeWeight and _getRatioWeight are similar, please check again.

**Review by mixed — APPROVED** [review]
LGTM

File: test/unit/PackingGrid.spec.ts [hunk 2]
```diff
@@ -106,4 +106,76 @@ describe("test PackingGrid", () => {
...
+
+    const maxInlinePos = Math.max(...items.map((item) => item.cssInlinePos + item.cssInlineSize));
+    const maxContentPos = Math.max(...items.map((item) => item.cssContentPos + item.cssContentSize));
+
+    expect(checks.every((chk) => chk)).to.be.true;
+    expect(maxInlinePos).to.be.closeTo(600, 0.000001);
+    expect(maxContentPos).to.be.closeTo(1200, 0.000001);
+    expect(container!.style.height).to.be.equals("1200px");
+  });
+  it(`should check if the distortion of the ratio is less when the weightPriority is "ratio" than "size"`, async () => {
```
Comment (WoodNeck) [inline test/unit/PackingGrid.spec.ts]: less...than what? Looks like the whole sentence is broken.

File: test/unit/PackingGrid.spec.ts [hunk 3]
```diff
@@ -106,4 +106,76 @@ describe("test PackingGrid", () => {
     expect(maxContentPos).to.be.closeTo(1200, 0.000001);
     expect(container!.style.height).to.be.equals("1200px");
   });
+  it(`should check if inlineSize:ContentSize is 1:2`, async () => {
```
Comment (WoodNeck) [inline test/unit/PackingGrid.spec.ts]: Can you add some conditions for the test names?
In this case, you should add in which condition inlineSize and contentSize should be 1:2

**Review by WoodNeck — APPROVED** [review]

**Pull request merged**